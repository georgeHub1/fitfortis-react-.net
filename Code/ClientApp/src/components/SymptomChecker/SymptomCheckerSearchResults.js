import React, { Component } from 'react';
import CustomCard from '../Common/CustomCard.js';
import { injectIntl, FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import styles from './styles.module.less';

export default injectIntl(
  class SymptomCheckerSearchResults extends Component {
    displayName = SymptomCheckerSearchResults.name;
    static propTypes = {
      tags: PropTypes.array
    };
    static defaultProps = {
      tags: []
    };
    render () {
      const {
        tags,
        causesAndDiagnosis,
        handleCloseModal,
        getBgColor
      } = this.props;

      return (
        <div className={styles.wrapCustomCard}>
          <CustomCard>
            <div className={styles.cardHeaderBlock}>
              <div className={styles.cardHeader}>
                <FormattedMessage id="SymptomCheckerSearchResults.causesAndDiagnoses" />
              </div>
            </div>
            <div className={styles.possibleCauses}>
              {tags.length > 0 ? (
                causesAndDiagnosis
                  .sort((a, b) => b.percent - a.percent)
                  .map(item => (
                    item.percent > 0 ? <div
                      key={item.name}
                      onClick={e => handleCloseModal(e, item)}
                      className={styles.itemCause}
                    >
                      <div
                        className={styles.percentBlock}
                        style={{ background: getBgColor(item.percent) }}
                      >
                        {item.percent}%
                      </div>
                      <div className={styles.nameCause}>{item.name}</div>
                    </div>
                    : null
                  ))
              ) : (
                <p className={styles.paragraphInfo}>
                  <FormattedMessage id="SymptomCheckerSearchResults.causesAndDiagnosesInfo" />
                </p>
              )}
            </div>
          </CustomCard>
        </div>
      );
    }
  }
);
