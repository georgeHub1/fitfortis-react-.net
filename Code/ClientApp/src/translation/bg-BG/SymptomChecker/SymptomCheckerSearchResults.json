{
    "causesAndDiagnoses": "Възможни диагнози", 
    "causesAndDiagnosesInfo": "Няма достатъчно данни за диагностициране. Добави повече от един симптом."
}