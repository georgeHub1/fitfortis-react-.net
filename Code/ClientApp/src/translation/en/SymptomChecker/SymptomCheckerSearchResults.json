{
    "causesAndDiagnoses": "Possible Causes and Diagnoses", 
    "causesAndDiagnosesInfo": "Not enough data to diagnose, try to add more than one symptom"
}