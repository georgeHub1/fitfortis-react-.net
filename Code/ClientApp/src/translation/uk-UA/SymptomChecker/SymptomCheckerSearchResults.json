{
    "causesAndDiagnoses": "Можливі причини та діагнози", 
    "causesAndDiagnosesInfo": "Недостатньо даних для діагностики, спробуйте додати кілька симптомів"
}