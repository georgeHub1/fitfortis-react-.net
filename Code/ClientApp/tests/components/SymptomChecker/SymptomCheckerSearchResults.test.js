import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import  SymptomCheckerSearchResults  from '../../../src/components/SymptomChecker/SymptomCheckerSearchResults.js';

const Container = SymptomCheckerSearchResults;

describe('SymptomCheckerSearchResults', () => {
  it('renders correctly', () => {
    const wrapper = renderer
      .create(
        <MemoryRouter>
          <Container />
        </MemoryRouter>
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
