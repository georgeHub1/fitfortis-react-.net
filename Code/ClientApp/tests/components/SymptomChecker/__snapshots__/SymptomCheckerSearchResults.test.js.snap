// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SymptomCheckerSearchResults renders correctly 1`] = `
<div>
  <div
    className="ant-card undefined customCardContainer ant-card-bordered"
    style={
      Object {
        "background": "rgba(255, 255, 255, 0.8)",
      }
    }
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div>
        <div />
      </div>
      <div>
        <p />
      </div>
    </div>
  </div>
</div>
`;
